### PR TITLE
Fix: Callbacks for process.stdout / .stderr

### PR DESCRIPTION
**Pull request by rluba**
The overrides for `process.stdout.write` and `process.stderr.write` did not invoke the provided callbacks when they’re done.

This solution is a bit hacky because it just waits for `stds.out` and `stds.err`, respectively. A proper fix should wait for all of the downstream writes (`std.std` and/or `std.out/err` and/or `process.send`, depending on the configuration), before invoking the callback – and forward any potential errors.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  |no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no, but the same four tests also fail on `development`
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: lib/ProcessContainer.js [hunk 1]
```diff
@@ -207,7 +207,7 @@ function exec(script, stds) {
         if (pm2_env.pm_err_log_path !== 'NULL' && pm2_env.pm_err_log_path !== '/dev/null' && stds.err.write) {
           stds.err.write(log_data, encoding, cb);
         }
-        else {
+        else if (cb) {
```
Comment (soyuka) [inline lib/ProcessContainer.js]: cb !== undefined

Comment (rluba) [inline lib/ProcessContainer.js]: Sure, I can change it, if you prefer that version. But the whole codebase seems to use boolean coercion all the time – just as I did.

Some examples:

```JavaScript
if (process.connected &&
    process.send &&
    process.versions &&
    process.versions.node)
        …
  });
```

or 

```JavaScript
if (… && stds.out.write) {
    …
}
```

Comment (soyuka) [inline lib/ProcessContainer.js]: yeah I know, just my 2 cents :). The PR looks fine to me anyway!

File: lib/ProcessContainer.js [hunk 2]
```diff
@@ -238,8 +241,11 @@ function exec(script, stds) {
         stds.std && stds.std.write && stds.std.write(log_data);
```
Comment (vmarchaud) [inline lib/ProcessContainer.js]: Should we change this to `write(log_data, encoding)` here too ?

Comment (soyuka) [inline lib/ProcessContainer.js]: we definitely should, nice catch!

Comment (rluba) [inline lib/ProcessContainer.js]: Thanks, done.

**Pull request merged**